src/app/concern/file-uploader/file-uploader.component.ts

```diff
@@ -1,18 +1,18 @@
 import {
   Component,
   ElementRef,
+  OnDestroy,
   OnInit,
-  ViewChild,
-  OnDestroy
+  ViewChild
 } from "@angular/core";
 import { FormBuilder, FormGroup } from "@angular/forms";
 import { Select, Store } from "@ngxs/store";
 import { Observable, Subscription } from "rxjs";
 import { SnackBarService } from "../../shared/services/snack-bar/snack-bar.service";
+import { IConcernSummary, IFileUploadProgress } from "../concern.interfaces";
 import { ACCEPTED_IMAGE_MIMES } from "../constants";
-import { Upload } from "../state/concern.actions";
+import { PrepareUpload } from "../state/concern.actions";
```

Comment:
This functionality was already written. Files In Progress already Means Files are in progress.

There was listener on selectedConcern which would execute the upload when an id is returned. I.e. concern has been saved.

Reply:
no worries feel free to update